src/cubi_tk/sodar_api.py

```diff
@@ -1,47 +1,32 @@
 import argparse
-from collections import namedtuple
 from functools import reduce
-import os
 from typing import Literal
 import urllib.parse as urlparse
 
 from loguru import logger
 import requests
 
-from .common import is_uuid, load_toml_config
+from cubi_tk.parsers import check_args_sodar_config_parser
+
+from .common import is_uuid
 from .exceptions import ParameterException, SodarAPIException
 
 
 class SodarAPI:
-    def __init__(self, sodar_url: str, sodar_api_token: str, project_uuid: str):
-        self.sodar_url = sodar_url
+    def __init__(self, sodar_server_url: str, sodar_api_token: str, project_uuid: str, config = None):
+        self.sodar_server_url = sodar_server_url
         self.sodar_api_token = sodar_api_token
         self.project_uuid = project_uuid
-        self.check_args()
+        self.check_args(config)
 
-    def check_args(self):
+    def check_args(self, config):
         # toml_config needs an object with attribute named config
-        args = namedtuple("args", ["config"])
-        toml_config = load_toml_config(args(config=None))
-        if not self.sodar_url:
-            if not toml_config:
-                raise ParameterException(
-                    "SODAR URL not given on command line and not found in toml config files."
-                )
-            self.sodar_url = toml_config.get("global", {}).get("sodar_server_url")
-            if not self.sodar_url:
-                raise ParameterException(
-                    "SODAR URL not found in config files. Please specify on command line."
-                )
-        if not self.sodar_api_token:
-            if not toml_config:
-                raise ParameterException(
-                    "SODAR API token not given on command line and not found in toml config files."
-                )
-            self.sodar_api_token = toml_config.get("global", {}).get("sodar_api_token")
-            if not self.sodar_api_token:
-                raise ParameterException(
-                    "SODAR API token not found in config files. Please specify on command line."
+        any_error, args= check_args_sodar_config_parser(argparse.Namespace(config=config, sodar_server_url=self.sodar_server_url, sodar_api_token= self.sodar_api_token))
+        self.sodar_server_url = args.sodar_server_url
+        self.sodar_api_token = args.sodar_api_token
+        if any_error:
+            raise ParameterException(
+                    "SODAR variables not found in config files. Please specify on command line."
                 )
         if not is_uuid(self.project_uuid):
             raise ParameterException("Sodar Project UUID is not a valid UUID.")
```

Comment:
I think this could just be moved to the `check_args_sodar_config_parser`, it should be either generally applicable or is probably not needed anymore.
While, the project_uuid part is not yet included in the "predefeined" sodar parser, it is however something the API will _always_ need and I don't think the sodar url and token can be used without a project uuid.
(We will also need some special handling for parsers that defined DESTINATIOn instead of project uuid, but I think a generalised check_args_sodar_config_parser (see also my comment there) can handle that.